src/core/execute.c

```diff
@@ -2196,6 +2196,54 @@ static int apply_working_directory(const ExecContext *context,
         return 0;
 }
 
+static int setup_keyring(Unit *u, const ExecParameters *p, uid_t uid, gid_t gid) {
+        key_serial_t keyring;
+
+        assert(u);
+        assert(p);
+
+        /* Let's set up a new kernel session keyring for all system services. This has the benefit that each service
+         * runs with its own keyring shared among all processes of the service. If we don't do this the keyring will be
+         * automatically created on-demand and then linked to the user keyring, we we'd rather avoid to have a
+         * per-service keyring. */
+
+        if (!(p->flags & EXEC_NEW_KEYRING))
+                return 0;
+
+        keyring = keyctl(KEYCTL_JOIN_SESSION_KEYRING, 0, 0, 0, 0);
+        if (keyring == -1) {
+                if (errno == ENOSYS)
+                        log_debug("Kernel keyring not supported, ignoring.");
+                else if (IN_SET(errno, EACCES, EPERM))
+                        log_debug("Kernel keyring access prohibited, ignoring");
+                else
+                        return log_debug_errno(errno, "Setting up kernel keyring failed: %m");
```

Comment:
note that the log messages are usually not visible anyway, as at that point in time we explicitly closed the logging fd already (in preparation for forking off the new process, as that process should not receive our logging fd). Hence you won't see them unless you patch systemd anyway, hence "debug" isn't totally off...

That said, it doesn't matter, and we can just use a proper log level here too, after all what you can't see you can't see. will fix.